### PR TITLE
[webtransport_h3_server] Support RFC version of HTTP Datagrams

### PR DESCRIPTION
**Pull request by bashi**
There is no significant semantics changes between draft04[1] and rfc[2].
* Capsule protocol was simlified.
* HTTP Setting value has changed.

[1] https://datatracker.ietf.org/doc/html/draft-ietf-masque-h3-datagram-04
[2] https://datatracker.ietf.org/doc/html/rfc9297

### CODE REVIEW

File: tools/webtransport/h3/webtransport_h3_server.py [hunk 1]
```diff
@@ -44,46 +45,49 @@
...
+
+
+class H3ConnectionWithDatagram(H3Connection):
     """
     A H3Connection subclass, to make it work with the latest
     HTTP Datagram protocol.
     """
-    H3_DATAGRAM_04 = 0xffd277
-    # https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-h3-websockets-00#section-5
+    # https://datatracker.ietf.org/doc/html/rfc9220#name-iana-considerations
```
Comment (nidhijaju) [inline tools/webtransport/h3/webtransport_h3_server.py:60]: nit: Should this be https://datatracker.ietf.org/doc/html/rfc9220#section-5-2.2.1 to be more specific?

File: tools/webtransport/h3/webtransport_h3_server.py [hunk 2]
```diff
@@ -146,47 +150,74 @@ def _h3_event_received(self, event: H3Event) -> None:
...
+    def _receive_h3_datagram_rfc_capsule_data(self, capsule: H3Capsule, fin: bool) -> None:
+        if capsule.type == CapsuleType.DATAGRAM_RFC:
+            raise ProtocolError(
+                f"Unimplemented capsule type: {capsule.type}")
+        elif capsule.type == CapsuleType.CLOSE_WEBTRANSPORT_SESSION:
+            self._set_close_info_and_may_close_session(
+                data=capsule.data, fin=fin)
+        else:
+            # Ignore unknown capsules.
+            pass
```
Comment (nidhijaju) [inline tools/webtransport/h3/webtransport_h3_server.py:180]: I think this should be `return` to be consistent with line 195.

**Review by nidhijaju — APPROVED** [review]
LGTM with nits :) Sorry for the delay!

**Comment by bashi** [comment]
I'm not sure why https://github.com/web-platform-tests/wpt/pull/39201/checks?check_run_id=12413093400 is happening. I think we disable the test on Firefox (https://github.com/web-platform-tests/wpt/blob/master/infrastructure/metadata/infrastructure/server/webtransport-h3.https.sub.any.js.ini)

@foolip @jgraham Do you have any idea?

**Pull request merged**